lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/client/UserCredentialSecurityTokenProvider.java

```diff
@@ -30,29 +33,75 @@
 /**
  * Reads security token from user credentials.
  */
-final class UserCredentialSecurityTokenProvider implements SecurityTokenProvider {
+public final class UserCredentialSecurityTokenProvider implements SecurityTokenProvider {
 
   private static final Logger LOG = Logger.getLogger(UserCredentialSecurityTokenProvider.class.getName());
 
   @Inject
-  private UserCredentialSecurityTokenProvider(){}
+  private UserCredentialSecurityTokenProvider() { }
 
   @Override
   public byte[] getTokens() {
+
     try {
+
       final UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
       final Credentials credentials = ugi.getCredentials();
+
+      LOG.log(Level.FINEST, "Got {0} tokens for user {1}", new Object[] {credentials.numberOfTokens(), ugi});
```

Comment:
Very good point! Here's what is being printed to the log:

    Added 1 tokens for user sergiym (auth:SIMPLE)

So I think we're safe :)